python/triton/language/semantic.py

```diff
@@ -1367,9 +1370,14 @@ def _validate_dtype(dtype, allowed_types, operand_name):
     assert lhs_rank == rhs_rank == 2 or lhs_rank == rhs_rank == 3, f"Both inputs must be either 2D or 3D; (lhs: {lhs.shape} vs rhs: {rhs.shape})"
     assert lhs.shape[-1].value == rhs.shape[
         -2].value, f"First input shape ({lhs.shape}) and second input shape {rhs.shape} are not compatible for matmul (second index of first shape ({lhs.shape[-1].value}) must be equal to first index of second shape ({rhs.shape[-2].value})"
-    assert lhs.shape[-2].value >= 16 and lhs.shape[-1].value >= 16 \
-        and rhs.shape[-1].value >= 16, \
-        f"All non-batch values in both first input shape ({lhs.shape}) and second input shape ({rhs.shape}) must be >= 16!"
+    if support_m16n16k8():
+        assert lhs.shape[-2].value >= 16 and lhs.shape[-1].value >= 8 \
+            and rhs.shape[-1].value >= 16, \
+            f"The first input shape MxN = {lhs.shape} and the second input shape NxK = {rhs.shape} must have M>=16, N>=16, K>=8 for arch gfx9x"
+    else:
+        assert lhs.shape[-2].value >= 16 and lhs.shape[-1].value >= 16 \
+            and rhs.shape[-1].value >= 16, \
+            f"All non-batch values in both first input shape ({lhs.shape}) and second input shape ({rhs.shape}) must be >= 16!"
     if lhs.type.scalar.is_int():
```

Comment:
the int type requirement is specific for cuda, so we should add remove it for amd backend.

Reply:
Yeah, I see. But I think this should go to a different PR. This PR concerns extending the `tl.dot` support to smaller sizes

Reply:
I now read https://github.com/triton-lang/triton/pull/4021, and yes, I can incorporate that change :)